### PR TITLE
chore(logstreams): fix append failure listener

### LINKED ISSUE #4084: LogStorageAppender  Incorrect usage of actor
**Describe the bug**

Found during investigating failed benchmark
```
java.lang.UnsupportedOperationException: Incorrect usage of actor. getLifecyclePhase(): must be called from actor thread at io.zeebe.util.sched.ActorControl.ensureCalledFromActorThread(ActorControl.java:522) ~[zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT] at io.zeebe.util.sched.ActorControl.ensureCalledFromWithinActor(ActorControl.java:507) ~[zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT] at io.zeebe.util.sched.ActorControl.getLifecyclePhase(ActorControl.java:498) ~[zeebe-util-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT] at io.zeebe.logstreams.impl.log.LogStorageAppender.onFailure(LogStorageAppender.java:199) ~[zeebe-logstreams-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT] at io.zeebe.logstreams.impl.log.LogStorageAppender$Listener.onWriteError(LogStorageAppender.java:233) ~[zeebe-logstreams-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT] at io.zeebe.logstreams.storage.atomix.AtomixAppendListenerAdapter.onWriteError(AtomixAppendListenerAdapter.java:29) ~[zeebe-logstreams-0.23.0-SNAPSHOT.jar:0.23.0-SNAPSHOT] at io.atomix.protocols.raft.roles.LeaderRole.safeAppendEntry(LeaderRole.java:1455) ~[atomix-raft-3.2.0-20200318.171657-141.jar:?] at io.atomix.protocols.raft.roles.LeaderRole.lambda$appendEntry$32(LeaderRole.java:1440) ~[atomix-raft-3.2.0-20200318.171657-141.jar:?] at io.atomix.utils.concurrent.SingleThreadContext$WrappedRunnable.run(SingleThreadContext.java:182) ~[atomix-utils-3.2.0-20200318.171617-141.jar:?] at io.atomix.utils.concurrent.SingleThreadContext$1.lambda$execute$0(SingleThreadContext.java:55) ~[atomix-utils-3.2.0-20200318.171617-141.jar:?] at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?] at java.util.concurrent.FutureTask.run(Unknown Source) [?:?] at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?] at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?] at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?] at java.lang.Thread.run(Unknown Source) [?:?]

```

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Use actor call to execute on failure method

## Related issues

closes #4084

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStorageAppender.java [hunk 1]
```diff
@@ -230,7 +230,7 @@ public void onWriteError(final Throwable error) {
         // Not a failure. It is probably during transition to follower.
         return;
       }
-      onFailure(error);
+      actor.run(() -> onFailure(error));
```
Comment (Zelldon) [inline logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStorageAppender.java]: speaks anything against moving the actor run into the onFailure?

Comment (deepthidevaki) [reply logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStorageAppender.java]: onFailure is also called in handleFailure(). If we use actor.run() there it won't be executed until other submitted tasks are executed. So we are not immediately failing when an unhandled exception happens.

Comment (Zelldon) [reply logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStorageAppender.java]: ok then leave it like it is thanks

**Review by ChrisKujawa — APPROVED** [review]
Thanks for this quick fix :+1:

**Pull request merged**